tests/test_hnsw_iterative_scan.py

```
@@ -0,0 +1,126 @@
+    # The function reads settings.DB.HNSW_ITERATIVE_SCAN at call time,
+    # so with monkeypatch it should return early without executing SQL.
+    db_module._set_hnsw_iterative_scan_on_connect(dummy_conn, None)  # pyright: ignore[reportPrivateUsage]
+    assert execute_calls == [], "No SQL should execute when HNSW_ITERATIVE_SCAN is None"
+
+
+def test_validate_pgvector_version_rejects_old_versions() -> None:
+    """_validate_pgvector_version raises RuntimeError for pgvector < 0.8.0."""
+    from src.db import _validate_pgvector_version  # pyright: ignore[reportPrivateUsage]
+
+    for old_version in ("0.7.0", "0.6.1", "0.5.0"):
+        with pytest.raises(RuntimeError, match="requires pgvector >= 0.8.0"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)